src/track/cueinfo.h

```diff
@@ -8,6 +8,12 @@
 
 namespace mixxx {
 
+enum class CueStatus {
```

Comment:
What is the need for adding this state in addition to the active/inactive state?

Reply:
Right now we have 2 states:
0 = Hotcue is not set
1 = Hotcue is set

With loop cues, we do need a third state:
2 = (Loop) cue is active

This is necessary for two things to represent this state on a controller, e.g. by letting the hotcue pad blink while the saved loop is active (we can't use another color since a cue can have an arbitrary color, which would make active loop cues indistinguishable from inactive loop cues with the same color).

The user needs to know whether the currently running loop is a saved loop or not, because if the former is the case that allows modifying the currently active saved loop, e.g. by using the loop shift/resize controls (otherwise you'd have to clear the loop and set it again first).

Reply:
But your comment got me thinking and I think it doesn't need to be stored inside the Cue object. It suffices if that status is local to `cuecontrol.cpp`.

Reply:
I like this refactoring.